packages/core/src/__tests__/state-manager.test.ts

```diff
@@ -348,16 +348,41 @@ describe('atCommit', () => {
       ceramic2._streamHandlers,
       snapshot.value
     )
+
+    // Snapshot is read-only
     await expect(snapshotStream.update({ abc: 1010 })).rejects.toThrow(
       'Historical stream commits cannot be modified. Load the stream without specifying a commit to make updates.'
     )
 
-    // Ensure that stateManager.atCommit does not mutate the passed in state object
-    expect(streamState2.state).toEqual(streamState2Original)
+    // We fast-forward streamState2, because the commit is legit
+    expect(streamState2.state).toEqual(stream1.state)
 
     await ceramic2.close()
   })
 
+  test('return read-only snapshot: do not lose own anchor status', async () => {
+    // Prepare commits to play
+    const tile1 = await TileDocument.create<any>(ceramic, INITIAL_CONTENT, null, {
+      anchor: false,
+      syncTimeoutSeconds: 0,
+    })
+    await tile1.update({ abc: 321, def: 456, gh: 987 })
+    await TestUtils.anchorUpdate(ceramic, tile1)
+
+    // Let's pretend we have a stream in PENDING state
```

Comment:
it would be weird to have a stream in anchorStatus PENDING where the most recent commit is an anchor commit.  Just to make the test slightly more realistic, I'd recommend dropping the `anchorUpdate` on line 370 and having one of the states have anchorStatus PENDING and the other have NOT_REQUESTED

Reply:
make sense, thanks